tasktiger/task.py

```diff
@@ -308,75 +307,26 @@ def _move(
         the expected queue.
         """
 
-        pipeline = self.tiger.connection.pipeline()
         scripts = self.tiger.scripts
-        _key = self.tiger._key
 
         from_state = from_state or self.state
         queue = self.queue
 
         assert from_state
         assert queue
 
-        scripts.fail_if_not_in_zset(
-            _key(from_state, queue), self.id, client=pipeline
-        )
-        if to_state:
-            if not when:
-                when = time.time()
-            if mode:
-                scripts.zadd(
-                    _key(to_state, queue), when, self.id, mode, client=pipeline
-                )
-            else:
-                pipeline.zadd(_key(to_state, queue), {self.id: when})
-            pipeline.sadd(_key(to_state), queue)
-        pipeline.zrem(_key(from_state, queue), self.id)
-
-        if not to_state:  # Remove the task if necessary
-            if self.unique:
-                # TODO: Do the following two in one call.
-
-                # Delete executions if there were no errors.
-                scripts.delete_if_not_in_zsets(
-                    to_delete=[
-                        _key("task", self.id, "executions"),
-                        _key("task", self.id, "executions_count"),
-                    ],
-                    value=self.id,
-                    zsets=[
-                        _key(state, queue) for state in {ERROR} - {from_state}
-                    ],
-                    client=pipeline,
-                )
-
-                # Only delete task if it's not in any other queue
-                check_states = {ACTIVE, QUEUED, ERROR, SCHEDULED} - {
-                    from_state
-                }
-                scripts.delete_if_not_in_zsets(
-                    to_delete=[_key("task", self.id)],
-                    value=self.id,
-                    zsets=[_key(state, queue) for state in check_states],
-                    client=pipeline,
-                )
-            else:
-                # Safe to remove
-                pipeline.delete(
-                    _key("task", self.id),
-                    _key("task", self.id, "executions"),
-                    _key("task", self.id, "executions_count"),
-                )
-
-        scripts.srem_if_not_exists(
-            _key(from_state), queue, _key(from_state, queue), client=pipeline
-        )
-
-        if to_state == QUEUED and self.tiger.config["PUBLISH_QUEUED_TASKS"]:
-            pipeline.publish(_key("activity"), queue)
-
         try:
-            scripts.execute_pipeline(pipeline)
+            scripts.move_task(
+                key_prefix=self.tiger.config["REDIS_PREFIX"],
+                id=self.id,
+                queue=self.queue,
+                from_state=from_state or self.state,
```

Comment:
`from_state` is sufficient since above you say `from_state = from_state or self.state`.